chain/network/src/peer/peer_actor.rs

```diff
@@ -844,21 +848,72 @@ impl StreamHandler<Result<Vec<u8>, ReasonForBan>> for PeerActor {
+                    last_time_received_message: ArcSwap::new(Arc::new(self.clock.now())),
+                    connection_established_time: self.clock.now(),
+                    throttle_controller: self.throttle_controller.clone(),
+                    send_accounts_data_demux: demux::Demux::new(
+                        self.network_state.send_accounts_data_rl,
+                    ),
+                }));
+
+                let connection_state = self.connection_state.clone().unwrap();
+                let tracker = self.tracker.clone();
+                let clock = self.clock.clone();
+                let period = self.network_state.config.peer_stats_period.try_into().unwrap();
+                actix::spawn(async move {
+                    loop {
+                        let sent = tracker.lock().sent_bytes.minute_stats(&clock);
+                        let received = tracker.lock().received_bytes.minute_stats(&clock);
+                        // TODO(gprusak): this stuff requires cleanup: only chain_info.height is
+                        // expected to change. Rest of the content of chain_info is not relevant
+                        // after handshake.
+                        connection_state.stats.store(Arc::new(Stats {
+                            received_bytes_per_sec: received.bytes_per_min / 60,
+                            sent_bytes_per_sec: sent.bytes_per_min / 60,
+                            // TODO(gprusak): deprecated, remove together with the Borsh encoding
+                            // support.
+                            encoding: None,
+                        }));
+                        // Whether the peer is considered abusive due to sending too many messages.
+                        // I am allowing this for now because I assume `MAX_PEER_MSG_PER_MIN` will
+                        // some day be less than `u64::MAX`.
+                        let is_abusive = received.count_per_min > MAX_PEER_MSG_PER_MIN
+                            || sent.count_per_min > MAX_PEER_MSG_PER_MIN;
+                        if is_abusive {
+                            trace!(target: "network", peer_id=?connection_state.peer_info.id, sent = sent.count_per_min, recv = received.count_per_min, "Banning peer for abuse");
+                            // TODO(MarX, #1586): Ban peer if we found them abusive. Fix issue with heavy
+                            //  network traffic that flags honest peers.
+                            // Send ban signal to peer instance. It should send ban signal back and stop the instance.
+                            // if let Some(connected_peer) = act.connected_peers.get(&peer_id1) {
+                            //     connected_peer.addr.do_send(PeerManagerRequest::BanPeer(ReasonForBan::Abusive));
+                            // }
+                        }
+                        tokio::time::sleep(period).await;
+                    }
+                });
```

Comment:
Am I correct that this isn't a leak, and that the task here would be killed when we stop the peer? A comment might be useful!

Reply:
actually, I don't know :). It calls directly tokio::task::spawn_local, so it might be totally detached from the actix system, however the thread is managed by the Arbiter handle, and Arbiters are stopped together with the SystemRunner, so maybe the task is dropped on SystemRunner::stop (stopping the actor is probably not enough). Anyway, I'll switch that to ctx.spawn(), so that it is clear that this is an ActorFuture.